### PR TITLE
KOGITO-1732 - Review serviceurl handling

### PR DESCRIPTION
**Pull request by cristianonicolai**
Assembly:
- https://github.com/kiegroup/kogito-runtimes/pull/425
- https://github.com/kiegroup/kogito-apps/pull/162

### CODE REVIEW

**Review by jstastny-cz — APPROVED** [review]

File: data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java [hunk 1]
```diff
@@ -140,11 +140,24 @@ private GraphQLSchema createSchema() {
...
+    protected String getProcessInstanceServiceUrl(DataFetchingEnvironment env) {
         ProcessInstance source = env.getSource();
+        if (source == null || source.getEndpoint() == null || source.getProcessId() == null) {
+            return null;
+        }
         String endpoint = source.getEndpoint();
-        String context = "/" + source.getProcessId();
-        return context.equals(endpoint) ? null : endpoint.substring(0, endpoint.indexOf(context));
+        LOGGER.debug("Process endpoint {}", endpoint);
+        if (endpoint.startsWith("/")) {
```
Comment (tiagodolphine) [inline data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:150]: it is worth checking if endpoint is `null`

File: data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java [hunk 2]
```diff
@@ -140,11 +140,24 @@ private GraphQLSchema createSchema() {
...
+            LOGGER.warn("Process '{}' endpoint '{}', does not contain full URL, please review the kogito.service.url system property to point the public URL for this runtime.",
+                        source.getProcessId(), endpoint);
+        }
+        String context = source.getProcessId().contains(".") ? source.getProcessId().substring(source.getProcessId().lastIndexOf('.') + 1) : source.getProcessId();
```
Comment (tiagodolphine) [inline data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java]: just suggestion to extract the `source.getProcessId().contains(".") ? source.getProcessId().substring(source.getProcessId().lastIndexOf('.') + 1) : source.getProcessId();` to a method.

**Review by tiagodolphine — COMMENTED** [review]
LGTM just minor comments.

File: data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java (hunk 1 above)
Comment (cristianonicolai) [reply data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:150]: I'm checking on line 145 :)

File: data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java (hunk 2 above)
Comment (cristianonicolai) [reply data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java]: done

File: data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java (hunk 1 above)
Comment (tiagodolphine) [reply data-index/data-index-service/src/main/java/org/kie/kogito/index/graphql/GraphQLSchemaManager.java:150]: +1

**Review by tiagodolphine — APPROVED** [review]

**Pull request merged**